softpack_core/schemas/environment.py

```diff
@@ -360,7 +358,29 @@ def create(cls, env: EnvironmentInput) -> CreateResponse:  # type: ignore
 
             version += 1
 
-        # Send build request
+        response = cls.submit_env_to_builder(env)
+        if response is not None:
+            return response
+
+        return CreateEnvironmentSuccess(
+            message="Successfully scheduled environment creation"
+        )
+
+    @classmethod
+    def submit_env_to_builder(
+        cls, env: EnvironmentInput
+    ) -> Union[None, BuilderError, InvalidInputError]:
+        """Submit an environment to the builder."""
+        try:
+            m = re.fullmatch(r"^(.*)-(\d+)$", env.name)
```

Comment:
Are we abandoning non-integer versions?

Reply:
What was our conclusion on this? something like:

- `submit_env_to_builder` can only ever get called for "real" SoftPack-built environments
- those environments have an autoincrementing integer version number (now that `gsb build` environments go via the core, it's not possible to manually specify a version number for those)
- so imported modules can still safely have a non-integer version number even with this change? (or, will be able to once dots are allowed in environment names)

Reply:
Looking at the code paths, that looks correct. It may be worth modifying the method comment to note that.